components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go

```diff
@@ -0,0 +1,134 @@
+package handlers
+
+import (
+	"database/sql"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/orchestration"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type kymaHandler struct {
+	db    storage.Orchestration
+	queue *process.Queue
+	log   logrus.FieldLogger
+}
+
+func NewKymaOrchestrationHandler(db storage.Orchestration, executor process.Executor, log logrus.FieldLogger) *kymaHandler {
+	return &kymaHandler{
+		db:    db,
+		log:   log,
+		queue: process.NewQueue(executor, log),
+	}
+}
+
+func (h *kymaHandler) AttachRoutes(router *mux.Router) {
+	router.HandleFunc("/orchestrations/{orchestration_id}", h.getUpgradeKymaOrchestrationsHandler).Methods(http.MethodGet)
+	router.HandleFunc("/kyma/upgrade", h.upgradeKymaOrchestrationHandler).Methods(http.MethodPost)
+}
+
+func (h *kymaHandler) getUpgradeKymaOrchestrationsHandler(w http.ResponseWriter, r *http.Request) {
+	orchestrationID := mux.Vars(r)["orchestration_id"]
+
+	o, err := h.db.GetOrchestrationByID(orchestrationID)
+	if err != nil {
+		h.log.Errorf("while getting orchestration %s: %v", orchestrationID, err)
+		writeErrorResponse(w, http.StatusInternalServerError, errors.Wrapf(err, "while getting orchestration %s", orchestrationID))
+	}
+
+	writeResponse(w, http.StatusOK, o)
+}
+
+func (h *kymaHandler) upgradeKymaOrchestrationHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
as before, consider renaming `Handlers` methods to more precise ones, something like:

```suggestion
func (h *kymaHandler) createOrchestration(w http.ResponseWriter, r *http.Request) {
```

`Handler` handles `upgradeKymaOrchestration` - in my opinion this is not very informative.
This method creates an `Orchestration` object which is passed to the queue to process. Let's try to figure out some more descriptive method names